proxy/src/schema.rs

```diff
@@ -54,11 +54,11 @@ impl Query {
     }
 
     fn projects(ctx: &Context) -> FieldResult<Vec<Project>> {
-        Ok(ctx.source.get_all_projects())
+        Ok(ctx.source.get_all_projects()?)
```

Comment:
This fails because the return type `FieldResult` only expects certain kinds of errors, namely ones to do with Juniper:

```
error[E0308]: mismatched types
  --> src/schema.rs:61:9
   |
61 |         ctx.source.get_project(id.into())
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `juniper::FieldError`, found enum `substrate_subxt::error::Error`
```

Somehow wrapping it in an Ok satisfied the compiler, which is either magical or concerning. I'll switch it out to use an enum of errors.

Reply:
Hmmm maybe this is something to do with `FieldResult`.

So [`FieldResult`](https://docs.rs/juniper/0.14.1/juniper/type.FieldResult.html) is just a type wrapper around `Result`. So I guess there's some magic going on with type conversions using `?`.

Sorry for the wild goose chase!